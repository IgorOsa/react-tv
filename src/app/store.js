import { configureStore } from '@reduxjs/toolkit';
import searchReducer, { SEARCH_FEATURE_KEY } from '../features/search/searchSlice';
import showsReducer, { SHOWS_FEATURE_KEY } from '../features/shows/showsSlice';

const store = configureStore({
  reducer: {
    [SHOWS_FEATURE_KEY]: showsReducer,
    [SEARCH_FEATURE_KEY]: searchReducer,
  },
});

// store.subscribe(() => {
//   console.log('store', JSON.stringify(store.getState()[SHOWS_FEATURE_KEY].status));
// });

export default store;
