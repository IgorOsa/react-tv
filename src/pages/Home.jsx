/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom/cjs/react-router-dom.min';
import { selectShows, fetchShowsAsync } from '../features/shows/showsSlice';
import './Home.scss';
import ShowCard from '../components/ShowCard';
import SearchBar from '../components/SearchBar';

function Home() {
  // const shows = useSelector((state) => selectTopShows(state, 12));
  const shows = useSelector(selectShows);
  const dispatch = useDispatch();

  // useEffect(() => {
  //   dispatch(fetchShowsAsync());
  // }, []);

  // const genres = new Set(shows.reduce((acc, el) => {
  //   acc.push(...el.genres);
  //   return acc;
  // }, []));

  // console.log([...genres].sort());

  return (
    <>
      <SearchBar />
      <Container>
        <Row>
          <Col className="text-center pt-4 pb-2">
            <h2>Shows</h2>
          </Col>
        </Row>
        <Row>
          {shows && shows.map(
            (el) => (
              <Col key={el.id} className="d-flex flex-row justify-content-center m-2 text-center">
                <Link to={`/show/${el.id}`} className="card__link">
                  <ShowCard name={el.name} image={el && el.image && el.image.medium} />
                </Link>
              </Col>
            ),
          )}
        </Row>
      </Container>
    </>
  );
}

export default Home;
