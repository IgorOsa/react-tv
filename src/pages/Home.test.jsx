import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../app/store';
import Home from './Home';

it('should renders <Home /> page correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
