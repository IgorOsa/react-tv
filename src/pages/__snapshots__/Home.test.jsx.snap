// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <Home /> page correctly 1`] = `
Array [
  <div
    className="search-bar"
  >
    <div
      className="container"
    >
      <div
        className="d-flex justify-content-center p-2 m-3 row"
      >
        <div
          className="col-12 col"
        >
          <form
            className=""
          >
            <div
              className="input-group input-group-sm"
            >
              <span
                className="input-group-text"
                id="inputGroup-sizing-sm"
              >
                Search
              </span>
              <input
                aria-describedby="inputGroup-sizing-sm"
                aria-label="Small"
                className="form-control"
                onChange={[Function]}
                placeholder="Type to search"
                role="search"
                type="search"
                value=""
              />
              <button
                aria-expanded={false}
                className="dropdown-toggle btn btn-secondary"
                disabled={false}
                id="dropdown-basic-button"
                onClick={[Function]}
                type="button"
              >
                Filter by Genre
              </button>
              <div
                className="dropdown-menu"
                style={Object {}}
                x-placement="bottom-start"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-center pt-4 pb-2 col"
      >
        <h2>
          Shows
        </h2>
      </div>
    </div>
    <div
      className="row"
    />
  </div>,
]
`;
